spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -151,13 +154,15 @@
       FileScanTask task,
       Schema readSchema,
       Map<Integer, ?> idToConstant) {
-    return Parquet.read(location)
-        .project(readSchema)
+    Parquet.ReadBuilder builder = Parquet.read(location)
         .split(task.start(), task.length())
+        .project(readSchema)
         .createReaderFunc(fileSchema -> SparkParquetReaders.buildReader(readSchema, fileSchema, idToConstant))
         .filter(task.residual())
-        .caseSensitive(caseSensitive)
-        .build();
+        .caseSensitive(caseSensitive);
+
+    return nameMapping != null ?
+        builder.withNameMapping(NameMappingParser.fromJson(nameMapping)).build() : builder.build();
```

Comment:
Same here, let's separate the name mapping from build.

Reply:
Fixed.